vhdbuilder/packer/cleanup-vhd.sh

```diff
@@ -1,9 +1,13 @@
 #!/bin/bash -eux
 
-# Cleanup packer SSH key and machine ID generated for this boot
-rm -f /root/.ssh/authorized_keys
-rm -f /home/packer/.ssh/authorized_keys
-rm -f /var/log/cloud-init.log /var/log/cloud-init-output.log 
-rm -f /etc/machine-id
-touch /etc/machine-id
-chmod 644 /etc/machine-id
\ No newline at end of file
+[ -z "${RUN_VHD_CLEANUP}" ] && echo "RUN_VHD_CLEANUP must be set for cleanup-vhd.sh to run" && exit 1
+
+if [ "${RUN_VHD_CLEANUP,,}" == "true" ]; then
```

Comment:
I would leave this as it was

you may want to shuffle the ssh key deletion for AIB steps tho